### DISCUSSION TITLE
¿Cómo nos conectamos por ssh a una máquina virtual desde Windows?

### DISCUSSION THREAD

**Discussion by labra** [Arquisoft/faq#55] [Q&A] [answered]
Para conectarse por ssh a una máquina virtual desde Windows, ¿Cuáles serían las instrucciones?

**Comment by labra** [comment]
Si estás utilizando un sistema Windows y tienes una clave privada en formato PEM, puedes utilizar PuTTY para conectarte a tu máquina virtual en Azure. Aquí tienes los pasos:

1. **Descargar e instalar PuTTY:**
   Si aún no tienes PuTTY instalado, puedes descargarlo desde el sitio web oficial: [PuTTY Download Page](https://www.chiark.greenend.org.uk/~sgtatham/putty/latest.html).

2. **Convertir la clave privada a formato PPK (si es necesario):**
   PuTTY utiliza un formato de clave diferente llamado PPK. Si tu clave privada está en formato PEM, necesitas convertirla a PPK. Puedes hacerlo utilizando PuTTYgen, que viene con la instalación de PuTTY.

   - Abre PuTTYgen.
   - En la barra de menú, selecciona `Conversions` y luego elige `Import key`.
   - Selecciona tu clave privada en formato PEM y haz clic en `Open`.
   - PuTTYgen te pedirá que guardes la clave en formato PPK. Haz clic en `Save private key` y guarda la clave con una extensión `.ppk`.

3. **Conectar con PuTTY:**
   - Abre PuTTY.
   - En la ventana de configuración:
      - En "Host Name (or IP address)", ingresa la dirección IP de tu máquina virtual.
      - En "Port", asegúrate de que esté configurado en 22 (que es el puerto SSH por defecto).
      - En "Connection type", selecciona `SSH`.
      - En la categoría `Connection > SSH > Auth`, carga la clave privada PPK que guardaste.
      - Haz clic en `Open` para iniciar la conexión.

4. **Iniciar sesión:**
   PuTTY te pedirá el nombre de usuario. Ingresa el nombre de usuario que utilizas para tu máquina virtual en Azure y presiona `Enter`. Luego, ingresa tu contraseña si es necesario.

Con estos pasos, deberías poder conectarte a tu máquina virtual en Azure desde tu sistema Windows utilizando PuTTY. Recuerda que la clave privada que uses debe coincidir con la clave pública asociada a tu máquina virtual en Azure.

NOTA: Las siguientes instrucciones fueron generadas con ChatGPT por Alex Fernández

**Comment by dariogmori** [comment, marked as answer]
You can also use MobaXTerm to connect via SSH by following this steps:

![imagen](https://github.com/Arquisoft/faq/assets/91892789/cffd8652-c1da-418e-ac24-593ae428164e)

![imagen](https://github.com/Arquisoft/faq/assets/91892789/e9b2a924-af12-418a-9e10-1ddbe5e9b655)

Done!
